### PR TITLE
51857 Implementation of a checkbox to change the form behavior

### PR DESCRIPTION
**Pull request by yeneastgate**
related to https://github.com/onOffice-Web-Org/oo-wp-plugin/issues/871
changed log:
- Added a checkbox to change the form behavior
- Added an option to display the form as a modal in the UI settings

### CODE REVIEW

**Comment by fredericalpers** [comment]
nach Rücksprache für v5.3 vorgezogen.

**Comment by fredericalpers** [comment]
Hi @yeneastgate Unfortunately, the feature is not working as expected. I have already reset the database version, but the checkbox still doesn't seem to influence the form’s behavior.

Is there anything else that might need updating or any files that should be modified to ensure the checkbox functions correctly?

Thank you!

**Comment by yeneastgate** [comment]
@fredericalpers To resolve the issue, the template must be changed, and only the user can perform this action. Please update the template and test it again.

**Comment by fredericalpers** [comment]
> @fredericalpers To resolve the issue, the template must be changed, and only the user can perform this action. Please update the template and test it again.

Hi @yeneastgate Thank you for pointing this out. I'm not entirely sure which specific adjustments need to be made to the template to resolve the issue. Could you please provide a screenshot and a more detailed explanation of the necessary changes? Thanks in advance for your assistance!

**Comment by yeneastgate** [comment]
@fredericalpers Show form as modal has been activated.

image
image

Show form as modal has been deactivated. 
image

In the video, I have shown the steps you can watch to understand better.

https://github.com/user-attachments/assets/813ae9c0-4878-4e60-88e0-b6ba6c72ff6c

**Comment by fredericalpers** [comment]
Hi @yeneastgate ,

It seems there might be a misunderstanding here. In the comments above, you mentioned changes in the template. Therefore we need a detailed guide, including screenshots or a video, to replicate these changes correctly.

At the moment, the feature does not work for us, no matter what steps we take. These changes should be available in the template.dist folder when uploading the zip file of this branch. Even when copying all the files to the personalized folder the feature does not work. Please provide clear instructions on how to proceed so we can ensure everything is working as intended.

Thank you for your support!

**Comment by yeneastgate** [comment]
@fredericalpers You can watch the video for a more detailed understanding. If any steps are missing or unclear, please let me know, and I will review them.

https://github.com/user-attachments/assets/0ae05d3c-c0b9-44b7-a3b2-e8f79f2de437

Here are the steps to follow. If they are not working, you can refer to the video:
1. Template Requirement: Only the ownerleadgeneratorform template has this functionality.
2. Backward Compatibility: If you are using a custom template like onoffice-personalized, you will need to update the ownerleadgeneratorform file.
3. Plugin Version Check: Ensure the plugin version is correct if you cannot activate or deactivate the checkbox in the backend.

Note: If the issue persists, please provide me with a public link where the shortcode form is used, and the ownerleadgeneratorform template is selected, so I can investigate and identify the cause.

**Comment by yeneastgate** [comment]
> @yeneastgate Thank you for your update! I noticed that while the checkbox now successfully changes the behaviour in the frontend, it's no longer possible to submit the forms. Specifically, no emails are being sent, and no data record are being created. The error code 185 ("Missing or invalid record ids") appears in the debug log. Could you please investigate and address this issue? Let me know if you need additional details or assistance!
> 
> [debug (23).log](https://github.com/user-attachments/files/17847437/debug.23.log)

@fredericalpers I will check it.

**Comment by yeneastgate** [comment]
@fredericalpers I tested again on my local and also on this link (https://testinstanz-fal.onofficeweb.com/formular-leadgenerator/) and could not reproduce the bug (the form was submitted successfully). If possible please provide me with a video of the bug reproducing so I can continue to investigate. Thanks!

**Comment by fredericalpers** [comment]
> @fredericalpers
> 
> 1. The test results with the user account you sent me in the email with the link https://testinstanz-fal.onofficeweb.com. Currently, I have deployed a form identical to the one on your faulty page, and I can still submit the form successfully. You can watch the video.

Has the form submission successfully triggered an email to be sent and created an address data record? I didn’t receive any email

File: plugin/FormPost.php [hunk 1]
```diff
@@ -131,7 +131,7 @@ public function initialCheck(DataFormConfiguration $pConfig, int $formNo)
 		} elseif ($pFormData->getMissingFields() !== []) {
 			$pFormData->setStatus(self::MESSAGE_REQUIRED_FIELDS_MISSING);
 			return;
-		}
+		}
```
Comment (mmilawski) [inline plugin/FormPost.php:134]: Leerzeichen kann entfernt werden

**Review by mmilawski — APPROVED** [review]

**Pull request merged**